### PR TITLE
disable always_false optimization in filter

### LINKED ISSUE #7164: TiFlash mpp does not compatible with `ConstantFilterDescription.always_false = true`
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
If `ConstantFilterDescription.always_false` is true, then `FilterBlockInputStream::readImpl()` will return `{}` immediately, which will stop currently `MPPTask`, and if the current mpp task has `ExchangeReceiver`, then it may stop the `ExchangeReceiver` unexpectedly, and when other mpp task is sending data to this task, it may meet error like 
```
DB::Exception: tunnel3+4: consumer exits unexpected, error message: Receiver cancelled, reason: Push mpp packet failed. Receiver state: CLOSED
```

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: close #7164

Problem Summary:

### What is changed and how it works?
Disable `always_false` optimization in Filter
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by yibin87 — APPROVED** [review]
LGTM

**Review by SeaRise — APPROVED** [review]
Others LGTM

File: dbms/src/Columns/FilterDescription.cpp [hunk 1]
```diff
@@ -30,12 +30,7 @@ extern const int ILLEGAL_TYPE_OF_COLUMN_FOR_FILTER;
 
 ConstantFilterDescription::ConstantFilterDescription(const IColumn & column)
 {
-    if (column.onlyNull())
-    {
-        always_false = true;
-        return;
-    }
-
+    // todo make always_false compatible with MPP
```
[the remark this answers is not in the document]
Comment (windtalker) [reply dbms/src/Columns/FilterDescription.cpp:33]: Done

**Comment by windtalker** [comment]
close as it does not solve the problem thoroughly, and for the specific problem we meet at daily tests in https://ci.pingcap.net/blue/organizations/jenkins/tiflash_regression_test_daily/detail/tiflash_regression_test_daily/4096/pipeline/
```
[2023-03-24T19:50:35.287Z]   File: /home/jenkins/agent/git/tiflash/integrated/ops/ti.sh.cmds/ci/fullstack_tests/randgen-mpp/mpp_subquery_join_1000.test

[2023-03-24T19:50:35.287Z]   Error line: 1759

[2023-03-24T19:50:35.287Z]   Error: tiflash use mpp_complex_sql1000; set @@tidb_isolation_read_engines='tiflash,tidb'; set @@tidb_allow_mpp=1;SELECT MAX( OUTR . col_datetime ) AS X FROM A AS OUTR2 LEFT JOIN A AS OUTR ON ( OUTR2 . col_datetime = OUTR . col_datetime_not_null AND OUTR2 . col_varchar_10_not_null > OUTR . col_varchar_10_not_null ) WHERE OUTR . col_decimal NOT IN ( SELECT INNR . col_decimal AS Y FROM AA AS INNR WHERE INNR . col_datetime_not_null < INNR . col_datetime OR NOT INNR . col_datetime_not_null > INNR . col_datetime ) GROUP BY OUTR . col_int HAVING X > 't';

[2023-03-24T19:50:35.287Z]   Result:

[2023-03-24T19:50:35.287Z]     ERROR 1105 (HY000) at line 1: other error for mpp stream: Can't find task [MPP<query:<query_ts:1679687418131966266, local_query_id:6803, server_id:2037375, start_ts:440319978529619976>,task_id:4>] within 10 s.

[2023-03-24T19:50:35.287Z]   Expected:

[2023-03-24T19:50:35.287Z]     <nothing>
```
it can be solved by https://github.com/pingcap/tiflash/pull/7175/files#diff-d720d0a6e1dd647d7541be568ee16f6945c4197aaf3f83eed2722d0b7a841855R418

**Pull request closed without merging**